functions/groups/businesses.js

```diff
@@ -0,0 +1,78 @@
+
+exports.getBusinesses = functions.https.onRequest(async (request, response) => {
+    var preCheck = await helpers.PreFunctionChecks(request, response, false, false);
+    if (preCheck.ret) return;
+
+    var state = request.body.data.state;
+    var city = request.body.data.city;
+    var category = request.body.data.category;
+
+    try {
+        var stores = db.collection(`businesses`);
+        if (state) {
+            stores = stores.where("state_key", "==", state);
+        }
+        if (city) {
+            stores = stores.where("city_key", "==", city);
+        }
+        if (category) {
+            stores = stores.where("category_key", "==", category);
+        }
+
+        stores = await stores.where("approved", "==", true).get();
+        var result = [];
+        stores.forEach((store) => {
+            var storeWithId = store.data()
+            storeWithId.id = store.id;
+            result.push(storeWithId)
+        });
+
+        response.send({ data: result });
+    } catch (e) {
+        response.status(500).send(`Error: ${e}`);
+    }
+    return;
+});
+
+exports.addBusiness = functions.https.onRequest(async (request, response) => {
```

Comment:
I agree that, long term, when a community member adds a business, it should go to the changes table. However, we will still need some way to move a biz from changes to biz table.

This is a good solution for now though

Reply:
That move from `changes` to `biz` will be part of the `changes` function group stuff. When some change is approved by an admin it will just read that change from `changes` and apply it.